pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -2004,6 +2026,22 @@ func TestReconcileClusterPool(t *testing.T) {
 				}
 			}
 
+			if test.expectedDeletionPossibleStatus != "" {
+				deletionPossibleCondition := controllerutils.FindCondition(pool.Status.Conditions, hivev1.ClusterPoolDeletionPossibleCondition)
+				if assert.NotNil(t, deletionPossibleCondition, "did not find DeletionPossible condition") {
+					assert.Equal(t, test.expectedDeletionPossibleStatus, deletionPossibleCondition.Status,
+						"unexpcted DeletionPossible condition status %s", deletionPossibleCondition.Status)
+				}
+			}
+
+			if test.expectedDeletionPossibleMessage != "" {
+				deletionPossibleCondition := controllerutils.FindCondition(pool.Status.Conditions, hivev1.ClusterPoolDeletionPossibleCondition)
+				if assert.NotNil(t, deletionPossibleCondition, "did not find DeletionPossible condition") {
+					assert.Equal(t, test.expectedDeletionPossibleMessage, deletionPossibleCondition.Message,
+						"unexpected DeletionPossible condition message %s", deletionPossibleCondition.Message)
+				}
+			}
+
```

Comment:
You can combine the logic here instead of fetching the condition twice.

Reply:
I looked at that. Because the two config fields are independent, the logic would end up being more convoluted and only slightly less repetitive :(

```suggestion
			if test.expectedDeletionPossibleStatus != "" || test.expectedDeletionPossibleMessage != "" {
				deletionPossibleCondition := controllerutils.FindCondition(pool.Status.Conditions, hivev1.ClusterPoolDeletionPossibleCondition)
				if assert.NotNil(t, deletionPossibleCondition, "did not find DeletionPossible condition") {
					if test.expectedDeletionPossibleStatus != "" {
						assert.Equal(t, test.expectedDeletionPossibleStatus, deletionPossibleCondition.Status,
							"unexpcted DeletionPossible condition status %s", deletionPossibleCondition.Status)
					}
					if test.expectedDeletionPossibleMessage != "" {
						assert.Equal(t, test.expectedDeletionPossibleMessage, deletionPossibleCondition.Message,
							"unexpected DeletionPossible condition message %s", deletionPossibleCondition.Message)
					}
				}
			}

```
Does that seem better to you? I'm okay either way.

I guess I could make the config field the whole condition and just check fields that are set... Yeah, yeah, I'll do that.